### PR TITLE
Added the flexibilty to skip waiting for the workload to complete to all workload fixtures

### PR DESCRIPTION
**Pull request by PrasadDesala**
Signed-off-by: Prasad Desala <tdesala@redhat.com>

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -3143,12 +3156,18 @@ def factory(
         log.info(f"Waiting for {waiting_time}sec to generate msg")
         time.sleep(waiting_time)
 
-        # Check messages are sent and received
-        threads = amq.run_in_bg(
-            namespace=kafka_namespace, value=value, since_time=since_time
-        )
+        if validate_messages:
```
Comment (mashetty330) [inline tests/conftest.py:3159]: what if only run_in_bg=True and validate_messages=False ? This condition doesnt seems to be correct

File: tests/conftest.py [hunk 2]
```diff
@@ -3143,12 +3156,18 @@ def factory(
...
+                threads = amq.run_in_bg(
+                    namespace=kafka_namespace, value=value, since_time=since_time
+                )
+                return amq, threads
+            else:
+                amq.validate_messages_are_produced()
+                amq.validate_messages_are_consumed()
 
-        return amq, threads
+        return amq
```
Comment (mashetty330) [inline tests/conftest.py:3171]: you may as well have to check all the tests for the respective workloads and make necessary changes

File: tests/conftest.py (hunk 1 above)
Comment (PrasadDesala) [reply tests/conftest.py:3163]: run_in_bg has to be called only when validate_messages is set to True. Now the code will throw exception for this case

File: tests/conftest.py (hunk 2 above)
Comment (PrasadDesala) [reply tests/conftest.py:3175]: Not needed as the newly introduced flags are by default set to True. I have plans to trigger the entire suite if I find any regressions will make the appropriate changes

**Review by OdedViner — APPROVED** [review]

**Review by mashetty330 — APPROVED** [review]

**Review by vavuthu — APPROVED** [review]

**Pull request merged**